### ISSUE TITLE
Issue with memcached servers retrieval

### ISSUE THREAD

**Issue by pierre-vigier** [PerlDancer/dancer2-session-memcached#6]
It seems theres an issue in file:
lib/Dancer2/Session/Memcached.pm

```perl
sub _build__memcached {
    my ($self) = @_;

    my $servers = $self->memcached_servers;

    croak "The setting memcached_servers must be defined"
      unless defined $servers;

    $servers = [ split /,\s*/, $servers ];

    # make sure the servers look good
    foreach my $s (@$servers) {
        if ( $s =~ /^\d+\.\d+\.\d+\.\d+$/ ) {
            croak "server `$s' is invalid; port is missing, use `server:port'";
        }
    }
```

That part of the code seems to assume that $self->memcached_servers will return string, as it's splitting it, however the coercion mechanism makes $self->memcached_servers return an array ref. The formatting is also taking care of during coercion.
Due to that, server IP and ports were not retrieve and i get a socket error, as no IP addresses are retrieved. Test are however passing succesfully, due to something add in the test suite.

If i install memcached locally, it will still use the mockup of the basic.t test. However, it i drop the memcached mockup in t/basic.t then the test are crashing, with the following error:

```
[App:73678] error @2018-06-06 18:21:47> Route exception: Unable to create a new session: Bad arg length for Socket::pack_sockaddr_in, length is 0, should be 4 at /Users/pierre.vigier/.plenv/versions/5.16.2/lib/perl5/5.16.2/darwin-thread-multi-2level/Socket.pm line 783.
 at /Users/pierre.vigier/.plenv/versions/5.16.2/lib/perl5/site_perl/5.16.2/Dancer2/Core/App.pm line 455. in /Users/pierre.vigier/.plenv/versions/5.16.2/lib/perl5/site_perl/5.16.2/Dancer2/Core/App.pm l. 1473
    not ok 1 - /set_session/larry
    #   Failed test '/set_session/larry'
    #   at t/basic.t line 124.
```

**Comment by pierre-vigier** [comment]
The pull request is fixing this bugs, however, i did not understand why test are passing with the current version.
To reproduce the bug, in t/basic.t, dropping the following

```perl
my $client = Cache::Memcached->new( servers => ["127.0.0.1:11211"] );

$client->set("hello", "world");

unless ( $client->get('hello') eq 'world'  ) {
    diag "no memcached server on localhost, will fake it";

    my %cache;

    no warnings 'redefine';

    sub Cache::Memcached::get    { $cache{$_[1]} }
    sub Cache::Memcached::set    { $cache{$_[1]} = $_[2] }
    sub Cache::Memcached::delete { delete $cache{$_[1]} }
}

# make sure we clean up from prior runs
$client->flush_all;
```
and installing memcached locally allow to reproduce

After the merge request, the test is passing again

**Comment by inigomerino** [comment]
I've confirmed this fixes a problem we were experiencing as well which made the module unusable in its present state. Any sense when this fix will make it to a new version of the module?

**Comment by yanick** [comment]
#7 merged and deployed as v0.007 to CPAN. Thanks!
